noir-r1cs/tests/compiler.rs

```diff
@@ -17,32 +13,10 @@ struct NargoTomlPackage {
     name: String,
 }
 
-fn run_nargo(path: impl AsRef<Path>) {
-    let status = std::process::Command::new("nargo")
-        .arg("compile")
-        .current_dir(path.as_ref())
-        .status()
-        .expect("Running nargo compile");
-
-    if !status.success() {
-        panic!("Failed to compile the test case");
-    }
-
-    let status = std::process::Command::new("nargo")
-        .arg("execute")
-        .current_dir(path.as_ref())
-        .status()
-        .expect("Running nargo execute");
-
-    if !status.success() {
-        panic!("Failed to execute the test case");
-    }
-}
-
 fn test_compiler(test_case_path: impl AsRef<Path>) {
     let test_case_path = test_case_path.as_ref();
 
-    run_nargo(test_case_path);
+    compile_workspace(test_case_path).expect("Compiling workspace");
```

Comment:
To clarify: It's of course fine to run `nargo` to do some stuff for end-to-end tests. And might even be preferred if it reduces dependency bloat or complexity.